File: src/service/shared/src/AgentsChatRoom/WebSockets/WebSocketLibrarianList.cs
using MultiAgents.WebSockets;
 

namespace MultiAgents.AgentsChatRoom.WebSockets
{

    public class WebSocektLibrainDocRef
    {
        public string? DocumentUri;
        public string? Text ;
        public string? Question;
        public string? Score;
    }

    public class WebSocketLibrarianList : WebSocketBaseMessage
    {
        public string? Question;
        public string? RoomName;
        public string? AgentName;
        public List<WebSocektLibrainDocRef> References = [];
      
    }
}

File: src/client/sample-app/src/models/WebSocketLibrarianList.ts
import { WebSocketBaseMessage } from './WebSocketBaseMessage';


/** Profile of an agent in a chat room */
export interface WebSocektLibrainDocRef {
    /** Agent's name */
    DocumentUri: string;
    Text: string;
    Question: string;
    Answer: string;
    Score: string;
  }
  

export interface WebSocketLibrarianList extends WebSocketBaseMessage {
    Question: string; // Base64 encoded audio data
    RoomName: string;
    AgentName: string;
    References: WebSocektLibrainDocRef[];
}
  